src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,620 @@
+    private Pattern nearbyTextPattern = Pattern.compile(DEFAULT_NEARBY_TEXT_PATTERN);
+
+    /**
+     * Specify check name pattern to suppress. Property can also be a RegExp group index
+     * at {@code nearbyTextPattern} in format of {@code $x} and be picked from line that
+     * matches {@code nearbyTextPattern}.
+     */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String checkPattern = DEFAULT_CHECK_PATTERN;
+
+    /** Specify check violation message pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String messagePattern;
+
+    /** Specify check ID pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String idPattern;
+
+    /**
+     * Specify negative/zero/positive value that defines the number of lines
+     * preceding/at/following the suppressing nearby text. Property can also be a RegExp group
+     * index at {@code nearbyTextPattern} in format of {@code $x} and be picked
+     * from line that matches {@code nearbyTextPattern}.
+     */
+    private String lineRange = DEFAULT_LINE_RANGE;
+
+    /** The absolute path to the currently processed file. */
+    private String currentFileAbsolutePath = "";
+
+    /**
+     * Setter to specify nearby text pattern to trigger filter to begin suppression.
+     *
+     * @param pattern a {@code Pattern} value.
+     */
+    public final void setNearbyTextPattern(Pattern pattern) {
+        nearbyTextPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check name pattern to suppress. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from line that matches {@code nearbyTextPattern}.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from line that matches {@code nearbyTextPattern}.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setLineRange(String format) {
+        lineRange = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+
+        if (event.getViolation() != null) {
```

Comment:
in what real case violation is null ?

Reply:
Not sure, to be honest. I was following what other filters are doing:

![Screenshot from 2023-03-19 16-37-27](https://user-images.githubusercontent.com/23459549/226187036-e23344bb-4e5d-4c86-bf4f-f0f633467cf5.png)

Reply:
Hmm, it is old mistake in design https://github.com/checkstyle/checkstyle/blob/8d0cf407779371f6c852abb7acc1ff56b1bfb7e2/src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java#L75
There might be idea to make it globally reusable, but in fact it is just for violations only. But conceptually it is possible to create event without violation. So filters should not be fragile if so.e Check create events in bad way